components/butterfly/src/rumor/election.rs

```diff
@@ -30,6 +30,24 @@ pub use protocol::newscast::{election::Status as ElectionStatus, Election as Pro
 use protocol::{self, newscast, FromProto};
 use rumor::{Rumor, RumorPayload, RumorType};
 
+// The default implementations leverage ElectionUpdate's Deref -> Election behavior, but this
+// generates a warning. In practice, it's fine:
+// See https://play.rust-lang.org/?version=stable&mode=debug&edition=2015&gist=162d64d74390df6a3a8bb815cd3a2c73
+#[allow(unconditional_recursion)]
+pub trait ElectionRumor {
```

Comment:
Since there are only two "subtypes" of election rumors, and you have to override this default implementation for `Election` anyway, how about removing these default implementations altogether in favor of concretely implementing them on each election type?

That eliminates this odd recursion warning, and also diminishes reliance on the `Deref` implementation for `ElectionUpdate` (Given that the docs say that `Deref` should only be implemented for "smart pointers", it feels like it's wrong to have it for `ElectionUpdate`, and by extension, wrong to build further upon it).

It's not really more verbose (you'll still have two implementations, after all), but it will be much more concrete and  leaves less open to potential weirdness in the future.

Reply:
Fair point. I want to get rid of the separate types at some point and have `Election` be an `enum` with `Leader` and `Update` variants, but until then, your suggestion is cleaner and simpler.

Reply:
That would be a nice future to get to, for sure 👍